packages/sdk-js/src/GrowthBook.ts

```diff
@@ -488,12 +500,37 @@ export class GrowthBook<
 
     // Apply new changes
     if (result.inExperiment) {
-      const undo = this._applyDOMChanges(result.value);
-      if (undo) {
-        this._activeAutoExperiments.set(experiment, {
-          undo,
-          valueHash,
-        });
+      if (result.value.urlRedirect) {
+        const url = experiment.persistQueryString
+          ? result.value.urlRedirect + window.location.search
+          : result.value.urlRedirect;
+        if (
+          experiment.urlPatterns &&
+          isURLTargeted(url, [experiment.urlPatterns[0]])
+        ) {
+          // log or throw error when redirect URL matches URL pattern?
+          return result;
+        }
+        this._redirectedUrl = url;
+        const navigate = this._getNavigateFunction();
+        if (navigate) {
+          this._setAntiFlicker();
```

Comment:
Move to inside default navigate function (prob not needed for something like next js